optimum/habana/transformers/generation/utils.py

```diff
@@ -1096,12 +1103,20 @@ def greedy_search(
             # prepare model inputs
             model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
 
+            hpu_graphs_kwargs = {}
+            if model_kwargs["limit_hpu_graphs"]:
+                hpu_graphs_kwargs.update({"bypass_hpu_graphs": False})
+                if "first_token" not in model_kwargs.keys():
+                    model_kwargs["first_token"] = True
+                    hpu_graphs_kwargs.update({"bypass_hpu_graphs": True})
```

Comment:
Maybe we could define a `get_hpu_graphs_kwargs` method in this class to factorize this block which is called in every decoding method?
So that we have:
```py
hpu_graphs_kwargs = get_hpu_graphs_kwargs(model_kwargs)
```
It will be easier to maintain.

Reply:
![image](https://github.com/huggingface/optimum-habana/assets/6461246/7bc890ae-54d1-4860-84f4-f1765ad32015)

i did actually but when i run style it changes the code which is wrong and i need to add on top of it something
i can unpack tuple then dict

Reply:
Oh that's weird! What do you return in `_get_hpu_graphs_kwargs`?